vars/cloudFoundryDeploy.groovy

```diff
@@ -293,9 +293,9 @@ void call(Map parameters = [:]) {
 
 private void handleMTADeployment(Map config, script) {
     // set default mtar path
-    config = ConfigurationHelper.newInstance(this, config)
-        .addIfEmpty('mtaPath', config.mtaPath ?: findMtar())
-        .use()
+    if(!config.mtaPath) {
```

Comment:
What was wrong with creating a new config helper and calling `addIfEmpty`? It does not make sense to discuss much here since the step will be moved to go soon. But in principle I think using a new builder for creating a new config is the better approach compared to manipulating an existing config. Just or the records: in general I think it would be better if the config map returned from `use()` would be immutable.

Reply:
Yes. I see your point and agree. 
We need if(!config.mtaPath) which was hidden before in `config.mtaPath  ?:` which I found a bit unintuive because otherwise findMtar() will always search for an mtar file and fail if multiple ones are found.

As I then already had `if(!config.mtaPath)` I thought addIfEmpty is a duplicate then. 

As it will anyway be removed. I will leave it as is for now.